src/bitops.c

```diff
@@ -16,6 +16,39 @@
 /* Count number of bits set in the binary array pointed by 's' and long
  * 'count' bytes. The implementation of this function is required to
  * work with an input string length up to 512 MB or more (server.proto_max_bulk_len) */
+#if defined(__GNUC__) || defined(__clang__)
+
+__attribute__((target("popcnt")))
+long long redisPopcount(void *s, long count) {
+    long long bits = 0;
+    unsigned char *p = s;
+    static const unsigned char bitsinbyte[256] = {0,1,1,2,1,2,2,3,1,2,2,3,2,3,3,4,1,2,2,3,2,3,3,4,2,3,3,4,3,4,4,5,1,2,2,3,2,3,3,4,2,3,3,4,3,4,4,5,2,3,3,4,3,4,4,5,3,4,4,5,4,5,5,6,1,2,2,3,2,3,3,4,2,3,3,4,3,4,4,5,2,3,3,4,3,4,4,5,3,4,4,5,4,5,5,6,2,3,3,4,3,4,4,5,3,4,4,5,4,5,5,6,3,4,4,5,4,5,5,6,4,5,5,6,5,6,6,7,1,2,2,3,2,3,3,4,2,3,3,4,3,4,4,5,2,3,3,4,3,4,4,5,3,4,4,5,4,5,5,6,2,3,3,4,3,4,4,5,3,4,4,5,4,5,5,6,3,4,4,5,4,5,5,6,4,5,5,6,5,6,6,7,2,3,3,4,3,4,4,5,3,4,4,5,4,5,5,6,3,4,4,5,4,5,5,6,4,5,5,6,5,6,6,7,3,4,4,5,4,5,5,6,4,5,5,6,5,6,6,7,4,5,5,6,5,6,6,7,5,6,6,7,6,7,7,8};
+    /* Count initial bytes not aligned to 64 bit. */
+    while((unsigned long)p & 7 && count) {
+        bits += bitsinbyte[*p++];
+        count--;
+    }
+
+#define ITER { \
+        bits += __builtin_popcountll(*(uint64_t*)(p + i)); \
+        i += 8; \
+    }
+
+    long i = 0;
+    while (i + 4*8 <= count) {
+        ITER ITER ITER ITER
+    }
+#undef ITER
```

Comment:
what about expand them directly? 
```suggestion
    /* Count bits 32 bytes at a time */ */
    while (i + 4*8 <= count) {
        bits += __builtin_popcountll(*(uint64_t*)(p + i));
        bits += __builtin_popcountll(*(uint64_t*)(p + i + 8));
        bits += __builtin_popcountll(*(uint64_t*)(p + i + 16));
        bits += __builtin_popcountll(*(uint64_t*)(p + i + 24));
        i += 32;
    }
```